modules/core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/DBCheck.java

```diff
@@ -0,0 +1,160 @@
+        }
+    }
+
+    /**
+     * Check driver availability and database connection
+     */
+    public void checkDatabaseConnection(ConfigurationHolder configHolder)
+            throws ConfigurationException, DatabaseDriverException, SQLException {
+        String databaseTemplate = configHolder.getIncludedDBTemplateName();
+        String dbName = configHolder.getProperty(PARAM_DB_NAME);
+        String dbUser = configHolder.getProperty(PARAM_DB_USER);
+        String dbPassword = configHolder.getProperty(PARAM_DB_PWD);
+        String dbHost = configHolder.getProperty(PARAM_DB_HOST);
+        String dbPort = configHolder.getProperty(PARAM_DB_PORT);
+
+        Path databaseTemplateDir = configHolder.getTemplatesPath().resolve(databaseTemplate);
+        String classname = configHolder.getProperty(PARAM_DB_DRIVER);
+        String connectionUrl = configHolder.getProperty(PARAM_DB_JDBC_URL);
+        // Load driver class from template or default lib directory
+        Driver driver = lookupDriver(configHolder, databaseTemplateDir, classname);
+        // Test db connection
+        DriverManager.registerDriver(driver);
+        Properties ttProps = new Properties();
+        ttProps.put(PARAM_DB_HOST, dbHost);
+        ttProps.put(PARAM_DB_PORT, dbPort);
+        ttProps.put(PARAM_DB_NAME, dbName);
+        ttProps.put(PARAM_DB_USER, dbUser);
+        ttProps.put(PARAM_DB_PWD, dbPassword);
+        TextTemplate tt = new TextTemplate(ttProps);
+        String url = tt.processText(connectionUrl);
+        Properties conProps = new Properties();
+        conProps.put("user", dbUser);
+        conProps.put("password", dbPassword);
+        log.debug("Testing URL: {} with: {}", url, conProps);
+        Connection con = driver.connect(url, conProps);
+        con.close();
+    }
+
+    /**
+     * Builds an {@link URLClassLoader} for the given databaseTemplate looking in the templates directory and in the
+     * server lib directory, then looks for a driver
+     *
+     * @param classname Driver class name, defined by {@link #PARAM_DB_DRIVER}
+     * @return Driver driver if found, else an Exception must have been raised.
+     * @throws DatabaseDriverException If there was an error when trying to instantiate the driver.
+     * @since 5.6
+     */
+    protected Driver lookupDriver(ConfigurationHolder configHolder, Path databaseTemplateDir, String classname)
+            throws DatabaseDriverException {
+        File[] files = ArrayUtils.addAll( //
+                databaseTemplateDir.resolve("lib").toFile().listFiles(), //
+                configHolder.getHomePath().resolve("lib").toFile().listFiles());
+        List<URL> urlsList = new ArrayList<>();
+        if (files != null) {
+            for (File file : files) {
+                if (file.getName().endsWith("jar")) {
+                    try {
+                        urlsList.add(new URL("jar:file:" + file.getPath() + "!/"));
```

Comment:
Could this be cleaned up following similar code on the NuxeoDeployer, using a PathMatcher to lookup libs? (https://github.com/nuxeo/nuxeo/blob/master/modules/runtime/nuxeo-runtime-tomcat-adapter/src/main/java/org/nuxeo/runtime/tomcat/NuxeoDeployer.java#L100)